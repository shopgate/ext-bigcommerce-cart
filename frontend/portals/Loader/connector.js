import { connect } from 'react-redux';
import { isCurrentViewLoading } from '@shopgate/pwa-common/selectors/view';

/**
 * Maps the contents of the state to the component props.
 * @param {Object} state The current application state.
 * @return {Object} The extended component props.
 */
const mapStateToProps = state => ({
  showLoadLoader: isCurrentViewLoading(state),
});

export default connect(mapStateToProps);
