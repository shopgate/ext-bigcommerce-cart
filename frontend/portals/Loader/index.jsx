import React from 'react';
import PropTypes from 'prop-types';
import connect from './connector';
import style from './styles';

/**
 * The Loader component.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Loader = props => (
  props.showLoadLoader ? <div className={style} /> : null
);

Loader.propTypes = {
  showLoadLoader: PropTypes.bool,
};

Loader.defaultProps = {
  showLoadLoader: false,
};

export default connect(Loader);
