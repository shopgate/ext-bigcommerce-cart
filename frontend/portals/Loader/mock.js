const PATHNAME = '/cart';

const history = {
  pathname: PATHNAME,
};

const activeIsLoading = {};
activeIsLoading[PATHNAME] = 1;

export const mockedStateNotLoading = {
  history,
  view: {
    isLoading: {},
  },
};

export const mockedStateLoading = {
  history,
  view: {
    isLoading: activeIsLoading,
  },
};

