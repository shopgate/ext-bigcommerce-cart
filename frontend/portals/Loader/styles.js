import { css } from 'glamor';

export default css({
  position: 'fixed',
  zIndex: 4,
  backgroundColor: 'rgba(0, 0, 0, 0.2)',
  width: '100vw',
  height: '100vh',
});
