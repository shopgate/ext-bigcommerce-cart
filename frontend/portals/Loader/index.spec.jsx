import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import mockRenderOptions from '@shopgate/pwa-common/helpers/mocks/mockRenderOptions';
import Loader from './index';
import { mockedStateNotLoading, mockedStateLoading } from './mock';

const mockedStore = configureStore();
const dispatcher = jest.fn();

beforeEach(() => {
  jest.resetModules();
});

describe('<Loader />', () => {
  let component = null;
  /**
   * Creates component with provided store state.
   * @param {Object} mockedState Mocked stage.
   * @param {Object} props Additional props.
   * @return {ReactWrapper}
   */
  const createComponent = (mockedState, props = {}) => {
    const store = mockedStore(mockedState);
    store.dispatch = dispatcher;

    return mount(
      <Provider store={store}>
        <Loader
          {...props}
        />
      </Provider>,
      mockRenderOptions
    );
  };

  beforeEach(() => {
    dispatcher.mockReset();
  });

  it('should only render when page is loading', () => {
    component = createComponent(mockedStateLoading);
    expect(component).toMatchSnapshot();
    expect(component.find('div').exists()).toBe(true);
  });

  it('should not render when page is not loading', () => {
    component = createComponent(mockedStateNotLoading);
    expect(component.find('div').exists()).toBe(false);
  });
});
