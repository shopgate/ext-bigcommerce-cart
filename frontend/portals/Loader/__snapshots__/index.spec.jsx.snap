// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should only render when page is loading 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Loader)>
    <Loader
      dispatch={[MockFunction]}
      showLoadLoader={true}
    >
      <div
        className={
          Object {
            "data-css-wkmi9o": "",
          }
        }
      />
    </Loader>
  </Connect(Loader)>
</Provider>
`;
